### ISSUE TITLE
Assertion `charBox && "Array unsupported in character lowering helper"' failed.

### ISSUE THREAD

**Issue by sscalpone** [flang-compiler/f18-llvm-project#509]
```
% cat int2a8.f
      SUBROUTINE INT 2 A8 (*,X,A8)
      CHARACTER*8     K8(1)
      REAL            X(1)
      GO TO 100
      ENTRY INT 2 K8 (*,X,K8)
 100  CONTINUE
      RETURN
      END
```

```
% bbc int2a8.f |& head
bbc: /local/home/sjs/mirror/fir-dev-mirror/flang/lib/Lower/CharacterExpr.cpp:126: fir::CharBoxValue Fortran::lower::CharacterExprHelper::toDataLengthPair(mlir::Value): Assertion `charBox && "Array unsupported in character lowering helper"' failed.
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
Stack dump:
0.      Program arguments: bbc int2a8.f
 #0 0x000055f00585e8fd llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/local/home/sjs/mirror/install/bin/bbc+0x345e8fd)
 #1 0x000055f00585c634 llvm::sys::RunSignalHandlers() (/local/home/sjs/mirror/install/bin/bbc+0x345c634)
 #2 0x000055f00585c7a0 SignalHandler(int) (/local/home/sjs/mirror/install/bin/bbc+0x345c7a0)
 #3 0x00007f2fda0a38a0 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x128a0)
 #4 0x00007f2fd8d0ff47 raise /build/glibc-2ORdQG/glibc-2.27/signal/../sysdeps/unix/sysv/linux/raise.c:51:0
 #5 0x00007f2fd8d118b1 abort /build/glibc-2ORdQG/glibc-2.27/stdlib/abort.c:81:0
```

**Comment by jeanPerier** [comment]
CharacterExprLower behing over-protective. In general, it's true that we do not know how to lower operations on array characters (say TRIM on an array), but processing F77 character argument, we absolutely can.

**Comment by jeanPerier** [comment]
Fixed by #514 merged in fir-dev.
